src/components/base/BaseButton.vue

```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts">
+import { computed, defineProps } from "vue";
+
+const props = defineProps({
+  buttonType: {
+    type: String,
+    default: "primary",
+  },
+  buttonSize: {
+    type: String,
+    default: "reg",
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
```

Comment:
We could actually omit this one, because it would be an attribute inherited by the button itself anyway (it's the root element of the template) 💡
Unless you would do something extra here with this prop, we can skip it 💪

Reply:
Great point, thank you